### DISCUSSION TITLE
明确inside world定位，

### DISCUSSION THREAD

**Discussion by Frepublic** [anobaka/Bakabase#296] [Ideas]
视频，图包，漫画，音声：目前漫画的管理看起来较为复杂

**Comment by anobaka** [comment]
这个标题一针见血，现在确实有点各方面都有但各方面都差一点的趋势。

因为我个人的资源涉及了多个领域，所以大方向上基本还是尽量涵盖多领域。

目前主要的问题是：最初设计的数据结构是仅为文件夹资源服务的，即1个文件夹对应1个资源，目前支持的单文件、压缩包都是基于此数据结构扩展而来的，导致现在至少有两个比较难处理的点：

**1. 不固定的目录结构，如**

a. 路径包含一部分信息的（如`年份`、`作者`）

```
2022年
  |----村上水军
          |----并没有作品
                   |----1.jpg
                   |----2.jpg
                   |----...
```

b. 路径包含非通用信息的

c. 资源名和分集名称不在一起的

```
葫芦娃A
  |----4K
       |----第一集.mp4
       |----.....
  |----1080P
       |----第一集.mp4

或

葫芦娃B
  |----第一集
         |----4K.mp4
         |----1080p.mp4
  |----....
```

以上述结构举例，会有几个问题：

1. 此时应该将`葫芦娃`视作一个资源，还是将`葫芦娃第一集`视作一个资源，还是将`葫芦娃4K第一集`视作一个资源？不同用户应该会有不同选择，需要有一个操作流程让用户可以方便地针对自己的喜好去配置；
2. 如果用户需要将1080P和4K视作资源的`分辨率`属性，那么针对上述`葫芦娃A和B`的情况，配置规则的方式应该如何设计才能让用户易于操作？
3. `分辨率`对于**音频资源**或**文本资源**来说几乎是无效信息，如果将`分辨率`设计为资源的`默认属性之一`，那在不断扩充新资源类型的同时，需要在**开发效率**、**架构复杂程度**和**性能**之间取一个平衡点；

d. 散图

```
散图包
   |----Yande爬虫
          |----aaa.jpg
          |----ccc.png
          |----数十万张图片.png
```

此时需要将`散图包A`视作资源还是`aaa.jpg`视作一个资源？该类资源可能需要单独的展示和操作交互，但也要注意和一般资源一起搜索时的兼容性

e. 压缩包存储方式

```
作者A
  |----资源A.zip
        |----1.jpg
        |----2.jpg
        |----...

作者A.zip
    |----资源A
           |----1.jpg
           |----2.jpg
           |----...
```

此时也应该有一个简单的操作流程能让用户配置`什么是资源`以及`资源的一些属性（如作者A）`

**2. 不同类型的资源，基础属性不同**

不同资源可能会有不同的基础属性，如：

+ 电影：年份、分辨率、演员等
+ 音声：码率等
+ 图片：分辨率等
+ 其他类型：...

需要避免属性膨胀、设计更为友好的搜索交互



基于以上两点，接下来初步确定的事情有：

**1. 简化现有媒体库配置交互，采用预置配置和询问的方式引导用户配置**

简易流程：
1. 选择存储的媒体类型：电影、动画、音声、图片、本子、自定义类型等
2. 选择根路径
3. 逐层扫描文件(夹)，询问每一层级是什么信息（如`2022年`对应`年份信息`）

**2. 优化部分架构**

1. 允许不同类型的资源有不同的默认属性、展示规则、搜索条件以及一些特殊能力
2. 将压缩包视作**文件夹**，采用和文件夹一样的处理方式
3. 资源增加父子关系，用于管理合集类资源

**3. 数据迁移**



剩下的想到再更新

**Comment by hyyin** [comment]
首先感谢一下大佬，偶然发现了此repository，简单了解后（还没来得及深入使用），觉得发现了宝藏。这正是心目中管理个人assets工具该有的样子（灵活敏捷，All in one，当然，UI 更美观就更好啦）。

本人目前使用 tinymediamanager + kodi 管理视频内容，体验还是不错的。tinymediamanager刮削后，可以选择将资源按特定结构重新整理。不知道这种思路对漫画等资源能否适用

总之，主要是表达一下对作者的敬佩和感谢，加油。

**Reply by anobaka** [reply]
感谢大佬的使用和建议，我这就去学习一下。
